PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/azure/hdinsight/spark/run/CosmosSparkBatchRunner.kt

```diff
@@ -52,18 +52,19 @@ class CosmosSparkBatchRunner : SparkBatchJobRunner() {
 
         val clusterId = submitModel.clusterId
         try {
-            val livyUri = submitModel.livyUri?.let { URI.create(it) } ?: AzureSparkCosmosClusterManager.getInstance()
-                    .findCluster(accountName, clusterId)
-                    .map { it.get().toBlocking().singleOrDefault(it).livyUri }
-                    .toBlocking()
-                    .firstOrDefault(null)
+            val clusterDetail = AzureSparkCosmosClusterManager.getInstance().findCluster(accountName, clusterId)
+                .toBlocking().singleOrDefault(null) ?: throw Exception("Can't get Spark on Cosmos cluster")
```

Comment:
`Exception` is too generic

Reply:
Will fix it in the next iteration

---
In reply to: [297084000](https://github.com/microsoft/azure-tools-for-java/pull/3201#discussion_r297084000) [](ancestors = 297084000)